common/src/main/scala/akka/contrib/persistence/mongodb/MongoReadJournal.scala

```diff
@@ -48,12 +76,60 @@ class JavaDslMongoReadJournal(rj: ScalaDslMongoReadJournal) extends javadsl.Read
     require(persistenceId != null, "PersistenceId must not be null")
     rj.currentEventsByPersistenceId(persistenceId, fromSequenceNr, toSequenceNr).asJava
   }
+  override def eventsByPersistenceId(persistenceId: String, fromSequenceNr: Long, toSequenceNr: Long) = {
+    require(persistenceId != null, "PersistenceId must not be null")
+    rj.eventsByPersistenceId(persistenceId, fromSequenceNr, toSequenceNr).asJava
+  }
+}
+
+trait JournalEventBus extends ActorEventBus with LookupClassification{
+  override protected def mapSize() = 65536
+
+  override protected def publish(event: Event, subscriber: Subscriber) = subscriber ! event
+
+  override protected def classify(event: Event) = event.pid
+
+  override type Classifier = String
+  override type Event = akka.contrib.persistence.mongodb.Event
+}
+
+trait JournalStream[A, Cursor] {
+    def cursor: Cursor
+    def publishEvent(handler: A => Unit): Unit
+    def streaming: Unit
+}
+
+class RemoveDuplicatedEvents extends PushStage[Event, Event]{
```

Comment:
Should this validate `persistenceId` is as expected?  I know that we *shouldn't* get the wrong one.